cmd/autoscaler/main.go

```diff
@@ -216,3 +231,24 @@ func flush(logger *zap.SugaredLogger) {
 	logger.Sync()
 	metrics.FlushExporter()
 }
+
+func componentConfig(ctx context.Context, logger *zap.SugaredLogger) leaderelection.ComponentConfig {
+	selfIP := os.Getenv("POD_IP")
```

Comment:
LookupEnv can help with more precise error message
But I'd rearrange the code to `leaderElectionConfig` and `cc` first.
Then just do `cc.Identity = os.LookupEnv`.

Reply:
I prefer to checking the env first and exit the process if the value is not set so that we don't need to do the job to set up leaderElectionConfig